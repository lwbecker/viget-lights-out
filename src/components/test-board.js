export const startingBoardOne = [
    [false, true, false, false, false],
    [false, false, false, false, false],
    [false, false, false, false, false],
    [false, false, true, false, false],
    [false, false, false, false, false],
];

const solveable = [
    [false, true, false, false, false],
    [true, true, true, false, false],
    [false, true, false, false, false],
    [false, false, false, false, false],
    [false, false, false, false, false],
];
