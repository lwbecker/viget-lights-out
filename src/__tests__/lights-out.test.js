import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import BoardContainer from '../containers/board-container.js';
import { startingBoardOne, solveable } from '../components/test-board.js';

afterEach(cleanup);

it('should start off with the correct board when passed a starting board', () => {
    const { getByTestId, asFragment } = render(
        <BoardContainer startingBoard={startingBoardOne} />
    );
    expect(asFragment()).toMatchSnapshot();
});

it('should flip appropriate lights when clicked', () => {
    const { getByTestId } = render(
        <BoardContainer startingBoard={startingBoardOne} />
    );

    //click Light at grid position 0-1
    fireEvent.click(getByTestId('0-1'));
    expect(getByTestId('0-0')).toHaveClass('on');
    expect(getByTestId('0-1')).toHaveClass('off');
    expect(getByTestId('0-2')).toHaveClass('on');
    expect(getByTestId('1-1')).toHaveClass('on');

    //click Light at grid position 2-3
    fireEvent.click(getByTestId('2-3'));
    expect(getByTestId('1-3')).toHaveClass('on');
    expect(getByTestId('2-2')).toHaveClass('on');
    expect(getByTestId('2-3')).toHaveClass('on');
    expect(getByTestId('2-4')).toHaveClass('on');
    expect(getByTestId('3-3')).toHaveClass('on');
});

it('should display winning message when all lights are off', () => {
    const { getByTestId } = render(
        <BoardContainer startingBoard={solveable} />
    );

    fireEvent.click(getByTestId('1-1'));
    expect(getByTestId('win-div')).toHaveTextContent(
        'Congratulations, you Won! Click reset to play again.'
    );
});

it('should remove winning message and reset moves to zero when the reset button is clicked', () => {
    const { getByTestId } = render(
        <BoardContainer startingBoard={solveable} />
    );

    fireEvent.click(getByTestId('1-1'));
    expect(getByTestId('win-div')).toHaveTextContent(
        'Congratulations, you Won! Click reset to play again.'
    );

    fireEvent.click(getByTestId('reset'));
    expect(getByTestId('win-div')).toBeUndefined();
});
