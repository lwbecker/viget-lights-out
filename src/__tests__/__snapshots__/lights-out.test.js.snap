// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should start off with the correct board when passed a starting board 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <div
      class="grid-container"
    >
      <button
        class="grid-item off"
        data-testid="0-0"
      />
      <button
        class="grid-item on"
        data-testid="0-1"
      />
      <button
        class="grid-item off"
        data-testid="0-2"
      />
      <button
        class="grid-item off"
        data-testid="0-3"
      />
      <button
        class="grid-item off"
        data-testid="0-4"
      />
      <button
        class="grid-item off"
        data-testid="1-0"
      />
      <button
        class="grid-item off"
        data-testid="1-1"
      />
      <button
        class="grid-item off"
        data-testid="1-2"
      />
      <button
        class="grid-item off"
        data-testid="1-3"
      />
      <button
        class="grid-item off"
        data-testid="1-4"
      />
      <button
        class="grid-item off"
        data-testid="2-0"
      />
      <button
        class="grid-item off"
        data-testid="2-1"
      />
      <button
        class="grid-item off"
        data-testid="2-2"
      />
      <button
        class="grid-item off"
        data-testid="2-3"
      />
      <button
        class="grid-item off"
        data-testid="2-4"
      />
      <button
        class="grid-item off"
        data-testid="3-0"
      />
      <button
        class="grid-item off"
        data-testid="3-1"
      />
      <button
        class="grid-item on"
        data-testid="3-2"
      />
      <button
        class="grid-item off"
        data-testid="3-3"
      />
      <button
        class="grid-item off"
        data-testid="3-4"
      />
      <button
        class="grid-item off"
        data-testid="4-0"
      />
      <button
        class="grid-item off"
        data-testid="4-1"
      />
      <button
        class="grid-item off"
        data-testid="4-2"
      />
      <button
        class="grid-item off"
        data-testid="4-3"
      />
      <button
        class="grid-item off"
        data-testid="4-4"
      />
    </div>
    <p>
      0
    </p>
    <button
      data-testid="reset"
    >
      Reset
    </button>
  </div>
</DocumentFragment>
`;
